.eslintrc.js

```diff
@@ -0,0 +1,61 @@
+/*
+👋 Hi! This file was autogenerated by tslint-to-eslint-config.
+https://github.com/typescript-eslint/tslint-to-eslint-config
+
+It represents the closest reasonable ESLint configuration to this
+project's original TSLint configuration.
+
+We recommend eventually switching this configuration to extend from
+the recommended rulesets in typescript-eslint. 
+https://github.com/typescript-eslint/tslint-to-eslint-config/blob/master/docs/FAQs.md
+
+Happy linting! 💖
+*/
+module.exports = {
+    "env": {
+        "es6": true
+    },
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "project": "tsconfig.json",
+        "sourceType": "module"
+    },
+    "plugins": [
+        "@typescript-eslint"
+    ],
+    "rules": {
+        "@typescript-eslint/await-thenable": "error",
```

Comment:
how different is this config from `@typescript-eslint/recommended`?
wondering if it's better to start from a basic config than this as it's hard to say which lint to keep or remove

Reply:
https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin has all the rules
@typescript-eslint/recommended which is based on syntax (no type checking) and
@typescript-eslint/recommended-requiring-type-checking which are based on semantics (type-checking)
 It is a superset of the rules we have in this config.. this config is what we had as tslint..

Reply:
Among all the typescript lint rules - the following we have which do not belong to recommended category (Key: :heavy_check_mark: = recommended)
@typescript-eslint/explicit-member-accessibility  @typescript-eslint/promise-function-async @typescript-eslint/member-ordering  @typescript-eslint/naming-convention @typescript-eslint/dot-notation 
There are several others - which appears as general rule not ts specific
"complexity": "error",
"curly": [
    "error",
    "multi-line"
],
"default-case": "error",
"eqeqeq": [
    "error",
    "always"
],
"import/no-extraneous-dependencies": "off",
"import/order": "error",
"no-console": "error",
"no-duplicate-imports": "error",
"no-magic-numbers": "error",
"no-return-await": "error",
"no-undef-init": "error",
"prefer-const": "error",
"prefer-template": "error",
"use-isnan": "error"

Reply:
Shouldn't there be an extend command like the following as mentioned https://github.com/eslint/eslint/tree/master/packages/eslint-config-eslint:
{
    "extends": "eslint"
}

Reply:
I am adding that as a separate PR - https://github.com/cennznet/api.js/pull/160 
Have extended the support for "plugin:@typescript-eslint/recommended" and in the process to extend
eslint:recommended and plugin:@typescript-eslint/recommended-requiring-type-checking in the same PR